software/py/func_call_log.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+
+# Script to generate function call log from PC trace and ELF file.
+# Usage: func_call_log.py <pc_trace_file> <elf_file>
+# 
+# PC trace format:
+# <line no>: <pc value in hex>
+# 
+# eg.:
+# 1:00000000
+# 2:00000001
+# ...
+# 234:0000fe45
+# ...
+#
+# Note: line numbers are prepended in the func log output so that 
+# they can be matched with correspoding line in the pc trace file.
+#
+# Bandhav Veluri
+# 7/31/19
+
+import sys
+import os
+import bisect
+
+if len(sys.argv) != 3:
+  print("Usage: func_call_log.py <pc_trace_file> <elf_file>")
+  sys.exit()
+
+
+###########################################
+# Parse ELF to extract function list
+
+# This works on any ELF
```

Comment:
Yes, but what does 'work' mean? You should comment what the sed command is doing (what are you searching and replacing).

Reply:
Isn't the comment "Parse ELF to extract function list" clear about what it's doing? I can elaborate and yes, that sed needs comments.

Reply:
That comment is good, but parsing a regex command in a human brain is always confusing. 
I guess it's easy to infer from the `['00000000 _start']` on the next line what the goal is. 
Here's a suggestion: instead of using sed for extracting two fields from stream, you should use awk. That would replace the sed command with `awk {print $2 " " $8}` which in awk speak means "print out the 2nd and 8th space separated word of each line." A little easier to understand than a regular expression.